apps/blog/bin/build-blog-files.sh

```
@@ -51,6 +51,42 @@ if ! node scripts/inject-home-jsonld.mjs; then
 fi
 
+# Storybook ("the workshop") ships alongside the site at /storybook/. It only
+# changes when components/tokens/stories/config do, so the slow webpack rebuild
+# is gated on a content hash of those inputs vs the hash stored beside the
+# deployed copy. Match -> restore the deployed Storybook (no rebuild). Mismatch
+# or unreadable bucket -> rebuild. out/storybook is ALWAYS populated; otherwise
+# prod-deploy.sh's `rsync -d` would delete the live workshop.
+SB_BUCKET="gs://kyle.pericak.com/storybook"
+SB_INPUTS=(.storybook components design-system lib package.json package-lock.json postcss.config.js tsconfig.json next.config.js)
+if command -v sha256sum >/dev/null 2>&1; then sha_cmd="sha256sum"; else sha_cmd="shasum -a 256"; fi
+sb_hash=$(find "${SB_INPUTS[@]}" -type f 2>/dev/null | LC_ALL=C sort | xargs $sha_cmd | $sha_cmd | awk '{print $1}')
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)